styx-flyte-client/src/main/java/com/spotify/styx/flyte/client/FlyteInputsUtils.java

```diff
@@ -73,24 +76,30 @@ private FlyteInputsUtils() {
                 LOG.error(message);
                 throw new IllegalArgumentException(message);
               } else {
-                literalMapBuilder.putLiterals(key, buildLiteralForPartition(parameter));
+                var literal = buildLiteralForPartition(parameter);
+                literalMapBuilder.putLiterals(key, literal);
               }
             });
 
     return literalMapBuilder.build();
   }
 
-  private static Timestamp toTimestamp(String value) {
-    Instant instant;
-    try {
-      instant = ParameterUtil.parseDate(value);
-    } catch (DateTimeParseException ignored1) {
+  static Timestamp toTimestamp(String value) {
+    List<Function<String, Instant>> functions = List.of(
```

Comment:
I would say that it would be better to have this code inside a method in `ParameterUtil`

```java
  public static Instant parseBest(String value) {
  }
```
Then we will add a lot of unit testing for this method and future refactoring or optimizations would be highly localized

Reply:
Will do this in a following PR